src/source_report_utils.py

```diff
@@ -0,0 +1,191 @@
+# functions used to generate source data reports
+import pandas as pd
+import urllib
+from src.constants import DATASET_BY_VERSION, DATASET_SOURCE_VERSIONS
+from src.digital_ocean_utils import (
+    OUTPUT_DATA_DIRECTORY_URL,
+    get_datatset_config,
+    get_data_library_sql_file,
+)
+
+from src.postgres_utils import (
+    QAQC_DB_SCHEMA_SOURCE_DATA,
+    create_sql_schema,
+    load_data_from_sql_dump,
+    get_schemas,
+    get_schema_tables,
+    get_table_columns,
+    get_table_row_count,
+)
+
+
+def dataframe_style_source_report_results(value: bool):
+    color = "rgba(0,155,0,.2)" if value else "rgba(155,0,0,.2)"
+    return f"background-color: {color}"
+
+
+def get_source_dataset_names(dataset: str, version: str) -> pd.DataFrame:
+    source_data_versions = get_source_data_versions_from_build(
+        dataset=dataset, version=version
+    )
+    return sorted(source_data_versions.index.values.tolist())
+
+
+def get_source_data_versions_from_build(dataset: str, version: str) -> pd.DataFrame:
+    try:
+        source_data_versions = pd.read_csv(
+            f"{OUTPUT_DATA_DIRECTORY_URL(dataset=dataset, version=version)}source_data_versions.csv",
```

Comment:
Not relevant to this PR but just a note - we don't have this norm on all data products. Some don't have source data at all, while edde uses a yaml file. Should probably settle on a norm and implement on all projects